blackbox/docs/sql/reference/alter_table.txt

```diff
@@ -112,3 +121,56 @@ columns with a value each to identify the partition to alter.
 :value: The columns value.
 
 .. SEEALSO:: :ref:`Alter Partitioned Tables <partitioned_tables_alter>`
+
+
+Sub-commands
+============
+
+``REROUTE``
+-----------
+
+The ``REROUTE`` command provides different options to allocate a shard on a node.
+It explictly enforces and executes a reroute allocation on the cluster and makes
+it possible ot move a shard from one node to another or even allocate
+unnassigned shards to a node or cancel allocations.
+
+::
+
+    [ REROUTE reroute_option ]
+
+:shard_id: The shard id. Ranges from 0 up to the specified number of shards of
+           a table.
+
+:node_id: The node ID within the cluster.
+
+
+``REROUTE`` suports the following options to start/stop shard allocation:
+
+``MOVE``
+~~~~~~~~
+
+A started shard gets moved from one node the another. It requests a
+``table_ident`` and a ``shard_id`` to identify the shard that receives the new
+allocation. Specify ``FROM node_id`` for the node to move the shard from and ``TO node_id``
+to move the shard to.
+
+``CANCEL``
+~~~~~~~~~~
+
+This cancels the allocation/recovery of a ``shard_id`` of a ``table_ident`` on a
+given ``node_id``. The ``allow_primary`` flag indicates if it is allowd to
+cancel the allocation of a primary shard.
+
+``ALLOCATE REPLICA``
+~~~~~~~~~~~~~~~~~~~~
+
+Allows to force allocation of an unassigned replica shard on a specific node.
+
+``RETRY FAILED``
+~~~~~~~~~~~~~~~~
+
+The setting ``index.allocation.max_retries`` indicates the maximum of attempts to
+to allocate a shard on a node. If this limit is reached it leaves the shard
+unallocated.
+This option triggers the allocation-retry of any unallocated shard that belong
+to the accepted ``table_ident``.
```

Comment:
i would provide examples and also explain when to use what why.

Reply:
@joemoe this is the reference page for `ALTER TABLE`. This only covers syntactical aspects. IMHO examples could be provided in the [DDLsection](https://crate.io/docs/crate/reference/en/latest/sql/ddl/alter_table.html) of our documentation.

Reply:
you are right. but i want to see examples. they should be at least linked here.